src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -706,36 +706,49 @@ public static ServerInfo GetServerVersion(SqlConnectionStringBuilder csb, string
                 azureAccountToken: azureAccountToken);
 
             return serverInfo;
-        }
-
-        /// <summary>
-        /// Gets the server host information from sys.dm_os_host_info view
-        /// </summary>
-        /// <param name="connection">The connection</param>
+        }
+
+        /// <summary>
+        /// Gets the server host information from sys.dm_os_host_info view
+        /// </summary>
+        /// <param name="connection">The connection</param>
         public static ServerHostInfo GetServerHostInfo(IDbConnection connection)
         {
-            // SQL Server 2016 and below does not provide sys.dm_os_host_info
+            var hostInfo = new ServerHostInfo();
+            // SQL Server 2016 and earlier versions does not provide sys.dm_os_host_info and we know the host OS can only be Windows.
             if (!Version.TryParse(ReadServerVersion(connection), out var hostVersion) || hostVersion.Major <= 13)
             {
-                return new ServerHostInfo
+                try
                 {
-                    Platform = "Windows"
-                };
-            }
-
-            var hostInfo = new ServerHostInfo();
-            ExecuteReader(
-                connection,
-                SqlConnectionHelperScripts.GetHostInfo,
-                reader =>
+                    hostInfo.Platform = "Windows";
+                    ExecuteReader(
+                        connection,
+                        SqlConnectionHelperScripts.GetHostWindowsVersion,
+                        reader =>
+                        {
+                            reader.Read();
+                            hostInfo.Release = reader[0].ToString();
+                        });
+                }
+                catch
                 {
-                    reader.Read();
-                    hostInfo.Platform = reader[0].ToString();
-                    hostInfo.Distribution = reader[1].ToString();
-                    hostInfo.Release = reader[2].ToString();
-                    hostInfo.ServicePackLevel = reader[3].ToString();
-                });
-
+                    // Ignore the error and only set the Platform to Windows by default
```

Comment:
may be a console log? just for ease of debugging in case...

Reply:
oops, I only saw the approved message and didn't see this one, I guess it is ok not to have it, at debug time, the intellisense will show the exception...